src/opendr/perception/semantic_segmentation/bisenet/CamVid.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2020-2021 OpenDR Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import glob
+import os
+from torchvision import transforms
+from PIL import Image
+import numpy as np
+from imgaug import augmenters as iaa
+import random
+import zipfile
+from urllib.request import urlretrieve
+from opendr.perception.semantic_segmentation.bisenet.algorithm.utils import get_label_info, RandomCrop, one_hot_it_v11
+from opendr.engine.datasets import ExternalDataset, DatasetIterator
+from opendr.engine.constants import OPENDR_SERVER_URL
+
+
+class CamVidDataset(ExternalDataset, DatasetIterator, torch.utils.data.Dataset):
+    """
+    CamVid dataset for Semantic Segmentation
+    path (dir): path directory of the CamVid dataset, containinig the subfolders "train",
+     "train_labels", "test", "test_labels", and the class_dict.csv file
+    scale (int): (crop_height, crop_width)
+    mode: 'test', 'train'
+    """
+
+    def __init__(self,
+                 path: str,
+                 scale=(720, 960),
+                 mode='train'):
+        ExternalDataset.__init__(self, path=str(path), dataset_type="camvid")
+        DatasetIterator.__init__(self)
+        torch.utils.data.Dataset.__init__(self)
+
+        self.mode = mode
+        if self.mode == 'train':
+            self.image_path = os.path.join(self.path, 'train')
+            self.label_path = os.path.join(self.path, 'train_labels')
+        elif self.mode == 'test':
+            self.image_path = os.path.join(self.path, 'test')
+            self.label_path = os.path.join(self.path, 'test_labels')
+        self.image_list = []
+        if not isinstance(self.image_path, list):
+            self.image_path = [self.image_path]
+        for image_path_ in self.image_path:
+            self.image_list.extend(glob.glob(os.path.join(image_path_, '*.png')))
+        self.image_list.sort()
+        self.label_list = []
+        if not isinstance(self.label_path, list):
+            self.label_path = [self.label_path]
+        for label_path_ in self.label_path:
+            self.label_list.extend(glob.glob(os.path.join(label_path_, '*.png')))
+        self.label_list.sort()
+        # self.image_name = [x.split('/')[-1].split('.')[0] for x in self.image_list]
+        # self.label_list = [os.path.join(label_path, x + '_L.png') for x in self.image_list]
```

Comment:
please add a comment explaining why these lines are commented out.
Same comment for other occurrences of commented lines in this and other newly added files.